cond.go

```diff
@@ -0,0 +1,253 @@
+}
+
+// Eq represents an equality comparison.
+func Eq(v interface{}) squirrel.Sqlizer {
+	return &condExpr{"=", v}
+}
+
+// Lt represents a less-than comparison.
+func Lt(v interface{}) squirrel.Sqlizer {
+	return &condExpr{"<", v}
+}
+
+// Lte represents a less-than-or-equal comparison.
+func Lte(v interface{}) squirrel.Sqlizer {
+	return &condExpr{"<=", v}
+}
+
+// IsNull represents an IS NULL comparison.
+func IsNull() squirrel.Sqlizer {
+	return &condExpr{"IS NULL", nil}
+}
+
+// IsNotNull represents an IS NOT NULL comparison.
+func IsNotNull() squirrel.Sqlizer {
+	return &condExpr{"IS NOT NULL", nil}
+}
+
+// Gt represents a greater-than comparison.
+func Gt(v interface{}) squirrel.Sqlizer {
+	return &condExpr{">", v}
+}
+
+// Gte represents a greater-than-or-equal comparison.
+func Gte(v interface{}) squirrel.Sqlizer {
+	return &condExpr{">=", v}
+}
+
+// NotEq represents a not-equal comparison.
+func NotEq(v interface{}) squirrel.Sqlizer {
+	return &condExpr{"<>", v}
+}
+
+// Like represents a LIKE comparison.
+func Like(v interface{}) squirrel.Sqlizer {
+	return &condExpr{"LIKE", v}
+}
+
+// ILike represents a ILIKE comparison.
+func ILike(v interface{}) squirrel.Sqlizer {
+	return &condExpr{"ILIKE", v}
+}
+
+// NotILike represents a NOT ILIKE comparison.
+func NotILike(v interface{}) squirrel.Sqlizer {
+	return &condExpr{"NOT ILIKE", v}
+}
+
+// NotLike represents a NOT LIKE comparison.
+func NotLike(v interface{}) squirrel.Sqlizer {
+	return &condExpr{"NOT LIKE", v}
+}
+
+// In represents an IN operator. The value must be variadic.
+func In(v ...interface{}) squirrel.Sqlizer {
+	return sqlExprFn(func() (string, []interface{}, error) {
+		if len(v) == 0 {
+			return "IN ()", nil, nil
+		}
+
+		return "IN (?" + strings.Repeat(", ?", len(v)-1) + ")", v, nil
+	})
+}
+
+// NotIn represents a NOT IN operator. The value must be variadic.
+func NotIn(v ...interface{}) squirrel.Sqlizer {
+	return sqlExprFn(func() (string, []interface{}, error) {
+		if len(v) == 0 {
+			return "NOT IN ()", nil, nil
+		}
+
+		return "NOT IN (?" + strings.Repeat(", ?", len(v)-1) + ")", v, nil
+	})
+}
+
+// AnyOf represents an IN operator. The value must be a slice.
+func AnyOf(v interface{}) squirrel.Sqlizer {
+	return sqlExprFn(func() (string, []interface{}, error) {
+		rv := reflect.ValueOf(v)
+
+		if rv.Kind() != reflect.Slice {
+			return "", nil, fmt.Errorf("value must be a slice")
+		}
+
+		if rv.Len() == 0 {
+			return "IN ()", nil, nil
+		}
+
+		vs := make([]interface{}, rv.Len())
+
+		for i := 0; i < rv.Len(); i++ {
+			vs[i] = rv.Index(i).Interface()
+		}
+
+		return "IN (?" + strings.Repeat(", ?", len(vs)-1) + ")", vs, nil
+	})
+}
+
+// NotAnyOf represents a NOT IN operator. The value must be a slice.
+func NotAnyOf(v interface{}) squirrel.Sqlizer {
+	return sqlExprFn(func() (string, []interface{}, error) {
+		rv := reflect.ValueOf(v)
+
+		if rv.Kind() != reflect.Slice {
+			return "", nil, fmt.Errorf("value must be a slice")
+		}
```

Comment:
Just a thought, maybe this could also take array.
```suggestion
// NotAnyOf represents a NOT IN operator. The value must be a slice or an array.
func NotAnyOf(v interface{}) squirrel.Sqlizer {
	return sqlExprFn(func() (string, []interface{}, error) {
		rv := reflect.ValueOf(v)

		rk := rv.Kind()
		if rk != reflect.Slice && rk != reflect.Array {
			return "", nil, fmt.Errorf("value must be a slice or an array")
		}
```

Reply:
same here `NotAnyOf[T](v []T)`

Reply:
IMHO This wouldn't work. You can't wrap it like this.
It would want the `[]interface{}` and it wouldn't accept `[]int`, `[]string`, etc.
`[]int != []interface{}`.
Sadly we don't have constraints for default non-primitve types yet (itf they would be ever created) only for primitives :/
https://pkg.go.dev/golang.org/x/exp@v0.0.0-20240325151524-a685a6edb6d8/constraints

Reply:
I think this is a great idea, we can remove `AnyOf` and replace it with a generic `In[T]` and remove reflection stuff. Doesn't look bad I think. The cost is creating a copy of the arguments, which is also necessary when using `AnyOf`, so I think it's acceptable. Please take a look at the most recent examples of `In` and `NotIn`.

Reply:
This is lovely :)

```
cond := pgkit.Cond{"list": pgkit.In[string]("Czech Republic", "Slovakia")}
```

which can be also written as
```
cond := pgkit.Cond{"list": pgkit.In("Czech Republic", "Slovakia")}
```